lang/mrustc/Portfile

```diff
@@ -59,46 +62,120 @@ compiler.blacklist *gcc*
+                OPENSSL_LIB_DIR=[openssl::lib_dir] \
+                RUSTC_VERSION=${rust_version} \
+                MRUSTC_TARGET_VER=${rust_version_major} \
+                "MINICARGO_FLAGS=--features curl-sys/force-system-lib-on-osx${legacy_features}" \
+                LIBGIT2_SYS_USE_PKG_CONFIG=1 \
+                OUTDIR_SUF=-${rust_version} \
+                "LLVM_CMAKE_OPTS_EXTRA=PYTHON_EXECUTABLE=${pythonfullpath} CMAKE_LIBTOOL=${prefix}/bin/libtool"
+
+build.target    -C run_rustc
+
 destroot {
-    xinstall -m 0755 -W ${worksrcpath} ./bin/${name} ${destroot}${prefix}/bin
-    xinstall -m 0755 -W ${worksrcpath} ./bin/minicargo ${destroot}${prefix}/bin
-    file copy ${worksrcpath}/output-${rust_version} ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    xinstall -m 0755 -W ${worksrcpath}/bin ${name} minicargo ${destroot}${prefix}/bin
+
+    # install all .rlib and .rlib.hir to lib/
+    xinstall -d ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    foreach f [glob -tails -directory ${worksrcpath}/output-${rust_version} lib*.rlib lib*.rlib.hir lib*.rlib.c lib*.rlib.o] {
+        xinstall -m 0644 -W ${worksrcpath}/output-${rust_version} $f \
+            ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    }
+
+    # keep boostraped rust
+    xinstall -d ${destroot}${prefix}/libexec/mrustc-${rust_version_major}/bin
+    foreach f {rustc cargo} {
+        xinstall -m 0755 -W ${worksrcpath}/output-${rust_version} $f \
+            ${destroot}${prefix}/libexec/mrustc-${rust_version_major}/bin
+        system -W ${destroot}${prefix}/bin/ "ln -s ../libexec/mrustc-${rust_version_major}/bin/$f ./mrustc-${f}"
+    }
+
+    # it needs its libs
+    file copy ${worksrcpath}/run_rustc/output-${rust_version}/prefix-2/lib \
+        ${destroot}${prefix}/libexec/mrustc-${rust_version_major}
+
+    system -W ${destroot}${prefix}/libexec/mrustc-${rust_version_major}/lib/rustlib/${configure.build_arch}-apple-darwin/lib \
+        "install_name_tool -id ${prefix}/libexec/mrustc-${rust_version_major}/lib/rustlib/${configure.build_arch}-apple-darwin/lib/libstd.dylib ./libstd.dylib"
+
+    system -W ${destroot}${prefix}/libexec/mrustc-${rust_version_major}/lib/rustlib/${configure.build_arch}-apple-darwin/lib \
+        "install_name_tool -id ${prefix}/libexec/mrustc-${rust_version_major}/lib/rustlib/${configure.build_arch}-apple-darwin/lib/libtest.dylib ./libtest.dylib"
+
+    system -W ${destroot}${prefix}/libexec/mrustc-${rust_version_major}/lib/rustlib/${configure.build_arch}-apple-darwin/lib \
+        "install_name_tool -change ${worksrcpath}/run_rustc/output-${rust_version}/build-std2/${configure.build_arch}-apple-darwin/release/deps/libstd.dylib ${prefix}/libexec/mrustc-${rust_version_major}/lib/rustlib/${configure.build_arch}-apple-darwin/lib/libstd.dylib ./libtest.dylib"
 }
 
 # Run a Hello, world! test
-test.env            RUSTC_VERSION=${rust_version} MRUSTC_LIBDIR=output-${rust_version}
```

Comment:
`MRUSTC_LIBDIR` was intentional because it needs to know where the built libraries are before they are installed. (Otherwise the `test` phase will look for libraries in `${prefix}`.)